docs/buildkit/index.md

```diff
@@ -0,0 +1,154 @@
+
+### Vagrant
+
+[Full Download: Vagrantbox](https://github.com/civicrm/civicrm-buildkit-vagrant) - Download a prepared virtual-machine with all system dependencies (mysql, etc). This is ideal if you work on Windows or OS X.
+
+
+### Docker
+
+If you have [Docker](https://www.docker.com/) running, you can [use this Docker image](https://github.com/ErichBSchulz/dcbk) to run buildkit.
+
+
+
+### Other platforms
+
+You may install buildkit in other environments. The main pre-requisites are:
+
+* Linux or OS X
+* Git
+* PHP 5.3+ (Extensions: `bcmath curl gd gettext imap intl imagick json mbstring mcrypt openssl pdo_mysql phar posix soap zip`)
+* NodeJS (v5 recommended)
+* NPM
+* Recommended (_for [amp](https://github.com/totten/amp) and [civibuild](/buildkit/civibuild.md)_)
+    * Apache 2.2 or 2.4 (Modules: `mod_rewrite`. On SUSE, possibly `mod_access_compat`. This list may not be exhaustive.)
+    * MySQL 5.1+ (client and server)
+
+All pre-requisites must support command-line access using the standard command
+names (`git`, `php`, `node`, `mysql`, `mysqldump`, etc). In some environments,
+you may need to enable these commands by configuring `PATH` -- this is especially
+true for MAMP, XAMPP, and other downloaded packages.
+(See, e.g., [Setup Command-Line PHP](http://wiki.civicrm.org/confluence/display/CRMDOC/Setup+Command-Line+PHP).)
+
+Once the pre-requisites are met, download buildkit to `~/buildkit`:
+
+```bash
+git clone https://github.com/civicrm/civicrm-buildkit.git ~/buildkit
+cd ~/buildkit
+./bin/civi-download-tools
+```
+
+### Troubleshooting
+
+* Nodejs version too old or npm update does not work
+
+Download the latest version from nodejs.org and follow to their instructions
+
+* Nodejs problems
+
+It might be handy to run
+
+```bash
+npm update
+npm install fs-extra
+```
+
+
+## Configuring buildkit after installation
+
+Buildkit includes many CLI commands in the `bin/` folder.
+
+You may execute the commands directly (e.g.  `./bin/civix` or `/path/to/buildkit/bin/civix`).  However, this would
+become very cumbersome.  Instead, you should configure the shell's `PATH` to recognize these commands automatically.
+
+!!! tip
+    Throughout this document, we will provide examples which assume that buildkit was downloaded to `/path/to/buildkit`. Be sure to adjust the examples to match your system.
+
+### Persistently add buildkit to `PATH` (typical)
+
+If you want to ensure that the buildkit CLI tools are always available, then:
+
+1. Determine the location of your shell configuration file. This is usually `~/.bashrc`, `~/.bash_profile`, or
+`~/.profile`.
+1. At the end of the file, add `export PATH="/path/to/buildkit/bin:$PATH"`
+1. Close and reopen the terminal.
+1. Enter the command `which civibuild`. This should display a full-path. If nothing appears, then retry the steps.
+
+### Temporarily switch shell configuration (advanced)
```

Comment:
This is a more atypical/advanced usage. In the interest of keeping TOC a bit smaller, would it make sense to combine these three headings ("Configuring buildkit after installation", " Persistently add buildkit to `PATH` (typical)", "Temporarily switch shell configuration (advanced)"). The main thread of the doc would discuss the persistent approach -- and then admonition could describe the advanced/temporary scenario?

Also: not sure what to do about this, but the PATH stuff should only apply to manual builds. For vagrant and docker, they should have PATH preset.